L1Trigger/VertexFinder/src/VertexFinder.cc

```diff
@@ -1161,20 +1162,26 @@ namespace l1tVertexFinder {
     std::map<int, float> nnOutput;
 
     float binWidth = 30./256.;
+    float temp_z0 = 0;
     // cout << " Filling inputPV" << endl;
     // Fill Histogram of 256 bins and input into NN
     for (float z = -15; z < 15.; z += binWidth) {
-      int zbin = (int)((z + 15.)/binWidth);
+      int zbin = floor((float)(((float)z + 15.)/((float)binWidth)));
       // int zbinPhalf = (int)(zbin+0.5*binWidth); // zbin plus half a binwidth
       float vxWeight = 0;
       for (const L1Track& track : fitTracks_) {
-        if (track.z0() >= z && track.z0() < (z + binWidth)) {
+        // Ad-hoc correction to z0 due to bias in +ve and -ve z0
+        if (track.z0() > 0.)
+          temp_z0 = track.z0() + 0.03;
```

Comment:
I miss the logic here...

Reply:
OK I guess we can keep it like this for now - but should keep it in mind moving forward